modules/rest/src/main/java/org/apache/ignite/rest/presentation/json/JsonConverter.java

```diff
@@ -35,6 +52,213 @@
+            for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+                String key = entry.getKey();
+
+                path.add(key);
+
+                JsonElement jsonElement = entry.getValue();
+
+                try {
+                    if (jsonElement.isJsonArray() || jsonElement.isJsonPrimitive())
+                        node.construct(key, new JsonPrimitiveConfigurationSource(path, jsonElement));
+                    else if (jsonElement.isJsonNull()) {
+                        node.construct(key, null);
+                    }
+                    else {
+                        assert jsonElement.isJsonObject();
+
+                        List<String> path = new ArrayList<>(this.path.size() + 1);
+                        path.addAll(this.path);
+                        path.add(key);
+
+                        node.construct(key, new JsonObjectConfigurationSource(path, jsonElement.getAsJsonObject()));
+                    }
+                }
+                catch (NoSuchElementException e) {
+                    throw new IllegalArgumentException(""); //TODO IGNITE-14372 Update comment.
+                }
+
+                path.remove(path.size() - 1);
+            }
+        }
+    }
+
+    private static class JsonPrimitiveConfigurationSource implements ConfigurationSource {
+        private final List<String> path;
+
+        private final JsonElement jsonLeaf;
+
+        private JsonPrimitiveConfigurationSource(List<String> path, JsonElement jsonLeaf) {
+            this.path = path;
+            this.jsonLeaf = jsonLeaf;
+        }
+
+        @Override public <T> T unwrap(Class<T> clazz) {
+            if (clazz.isArray() != jsonLeaf.isJsonArray())
+                throw new IllegalArgumentException(""); //TODO IGNITE-14372 Update comment.
+
+            return null;
+        }
+
+        @Override public void descend(ConstructableTreeNode node) {
+            throw new IllegalArgumentException(""); //TODO IGNITE-14372 Update comment.
+        }
+
+        private <T> T unwrap(JsonPrimitive jsonPrimitive, Class<T> clazz) {
+            assert !clazz.isArray();
+
+            if (clazz == String.class) {
+                if (!jsonPrimitive.isString())
+                    throw new IllegalArgumentException("");
+
+                return clazz.cast(jsonPrimitive.getAsString());
+            }
+
+            if (Number.class.isAssignableFrom(clazz)) {
+                if (!jsonPrimitive.isNumber())
+                    throw new IllegalArgumentException("");
+
+                if (clazz == Double.class)
+                    return clazz.cast(jsonPrimitive.getAsDouble());
+
+                if (clazz == Long.class)
+                    return clazz.cast(jsonPrimitive.getAsLong());
+
+                if (clazz == Integer.class) {
+                    long longValue = jsonPrimitive.getAsLong();
+
+                    if (longValue < Integer.MIN_VALUE || longValue > Integer.MAX_VALUE)
```

Comment:
you can use `Math.toIntExact` instead

Reply:
Ok, but then I would need to handle the exception) I asked about this exact thing at friday in the office)

Reply:
Why do you need to handle the exception? Anyway, I think that re-throwing an exception is better (at least they have a better cause message =))

Reply:
In the end this message will be visible to the user. I want him to see a proper text, not "integer overflow".

Reply:
These empty strings will be replaced with a meaningful messages in IGNITE-14372

Reply:
ok, got it!